reactioncommerce/reaction
# feat: option to choose whether a product should appear in the sitemap

Resolves #4705 
Impact: **minor**  
Type: **feature**

## Issue
There was no way to choose which products appeared in the generated sitemap.xml file. All published products were thus included.

## Solution
This PR adds a checkbox to product settings in the PDP admin panel labelled "Include in sitemap?". If checked, and the product is published, it will be included in the sitemap. If the product is already published when the checkbox is changed, a popup will appear asking the operator if he/she would like to regenerate the sitemap now. If yes is chosen, the sitemap will be regenerated. This PR also converts the `sitemaps/generate` meteor method to a GraphQL mutation named `generateSitemaps`

## Breaking changes
None


## Testing
1. Confirm migration success: in DB, Migrations.version should be 41, and locked should be false. Confirm all products have the "Included in sitemap?" checkbox checked via PDP -> Product Settings.
2. Visit shop options and refresh the sitemap. Confirm all published products are included in the sitemap (sitemap.xml -> sitemap-products-1.xml)
3. Uncheck "Include in sitemap?" for a published product. Confirm a popup that regenerates the sitemap when you select "Yes, regenerate now". Confirm that product is no longer in the sitemap
4. Uncheck "Include in sitemap?" for all published products, and regenerate the sitemap. Confirm there is no longer a sitemap-products-1.xml linked from sitemap.xml (and that it no longer exists if you visit it).
5. Create a new product and confirm "Include in sitemap?" is checked by default. Publish it, regenerate the sitemap, and confirm it's included in the sitemap.
